### DISCUSSION TITLE
GPTCache is ignoring cached question for the same session

### DISCUSSION THREAD

**Discussion by Sebastian7C** [zilliztech/GPTCache#583] [Q&A]
Hello,

I'm facing an issue with storing my cache by session. When I use the same session for a question that has already been cached, it seems like the cache is completely ignored. Here's a snippet of my code:

`from gptcache.session import Session
from gptcache import cache
from gptcache.adapter import openai
import os

cache.init()
cache.set_openai_key()

# Creating a session
session = Session(name="my-session")

# Asking a question and getting a response
response = openai.ChatCompletion.create(
    model='gpt-3.5-turbo',
    messages=[
        {
            'role': 'user',
            'content': "what's Github"
        }],
    session=session,
    temperature=0.9
)

# Asking the same question again with the same session
response = openai.ChatCompletion.create(
    model='gpt-3.5-turbo',
    messages=[
        {
            'role': 'user',
            'content': "what's Github"
        }],
    session=session,
    temperature=0.9
)
`
I'm wondering if anyone has insights on how to make sessions work properly in this context. Any help would be appreciated. Thanks!

**Comment by SimFG** [comment]
the test example maybe help you to solve the problem
https://github.com/zilliztech/GPTCache/blob/acc20f05400dabdcde451194e9bb73b986747685/tests/unit_tests/test_session.py#L83-L122

**Reply by SimFG** [reply]
doc: https://github.com/zilliztech/GPTCache/blob/main/examples/README.md#How-to-run-with-session
